gmso/formats/mol2.py

```diff
@@ -0,0 +1,177 @@
+                element = None
+            elif element_by_symbol(line[5]):
+                element = element_by_symbol(line[5])
+            elif element_by_name(line[5]):
+                element = element_by_name(line[5])
+            else:
+                warnings.warn(
+                    "No element detected for site {} with index{}, consider manually adding the element to the topology".format(
+                        line[1], len(topology.sites) + 1
+                    )
+                )
+                element = None
+            try:
+                charge = float(line[8])
+            except IndexError:
+                warnings.warn(
+                    "No charges were detected for site {} with index {}".format(
+                        line[1], line[0]
+                    )
+                )
+                charge = None
+            atom = Atom(
+                name=line[1],
+                position=position.to("nm"),
+                charge=charge,
+                element=element,
+                residue_label=line[7],
+                residue_index=int(line[6]),
+            )
+            topology.add_site(atom)
+        else:
+            break
+    return line
+
+
+def load_top_bonds(f, topology, **kwargs):
+    """Take a mol2 file section with the heading '@<TRIPOS>BOND' and save to the topology.bonds attribute."""
+    while True:
+        line = f.readline()
+        if "@" not in line and not line == "\n" and line:
+            line = line.split()
+            bond = Bond(
+                connection_members=(
+                    topology.sites[int(line[1]) - 1],
+                    topology.sites[int(line[2]) - 1],
+                )
+            )
+            topology.add_connection(bond)
+        else:
+            break
+    return line
+
+
+def load_top_box(f, topology, **kwargs):
+    """Take a mol2 file section with the heading '@<TRIPOS>FF_PBC' or '@<TRIPOS>CRYSIN' and save to topology.box."""
+    if topology.box:
+        warnings.warn(
+            "This mol2 file has two boxes to be read in, only reading in one with dimensions {}".format(
+                topology.box
+            )
+        )
+        line = f.readline()
+        return line
+    while True:
+        line = f.readline()
+        if "@" not in line and not line == "\n":
+            line = line.split()
+            # TODO: write to box information
+            topology.box = Box(
+                lengths=[float(x) for x in line[0:3]] * u.Å,
+                angles=[float(x) for x in line[3:6]] * u.degree,
+            )
+        else:
+            break
+    return line
+
+
+def parse_record_type_indicator(f, line, topology, site_type):
+    """Take a specific record type indicator from a mol2 file format and save to the proper attribute of a gmso topology.
+
+    Supported record type indicators include Atom, Bond, FF_PBC, and CRYSIN.
+    """
+    supported_rti = {
+        "@<TRIPOS>ATOM\n": load_top_sites,
```

Comment:
Assuming that these lines match exactly might not be the most robust. A suggestion here is to remove these `\n`. And change the conditional predicates to
`line.strip().startswith`

Reply:
In favor of this too. If the specification does not allow anything else, then this is probably fine. Can people add comments or anything in this file? If so, what umesh suggests will make it robust enough

Reply:
I removed the \n for the record types and added strip() methods to try to isolate just the record type indicator titles. The specification is pretty exact on how these titles should be named, so we should be okay here I think.

As for comments, comments are absolutely fine! As long as they're not embedded in the data, they shouldn't be read in. See `gmso/utils/files/benzene_ua.mol2` for an example file that has a generated by VMD comment at the end.